indexing-solr/src/main/java/edu/unc/lib/boxc/indexing/solr/filter/SetDatastreamFilter.java

```diff
@@ -282,6 +282,20 @@ private void addDatastreams(List<Datastream> dsList, List<BinaryObject> binList,
             });
     }
 
+    /**
+     * Used to selectively add the JP2 thumbnail datastream
+     *
+     * @param workObject the work object with the thumbnail relation
+     * @param datastreams work object's datastreams to add thumbnail streams to
+     */
+    private void addThumbnailDerivative(WorkObject workObject, List<Datastream> datastreams) {
```

Comment:
If we move this down to after "addDerivatives" where `addThumbnailDerivatives` was then this will look like a much smaller change